text/0000-builder-spec.md

```diff
@@ -0,0 +1,270 @@
+{
+  "description": "<Some description>",
+  "stack": {
+    "runImage": {
+      "image": "<some/run-image>",
+      "mirrors": [
+        "<gcr.io/some/default>"
+      ]
+    }
+  },
+  "buildpacks": [
+    {
+      "id": "<buildpack id>",
+	  "version": "<buildpack version>",
+	  "homepage": "http://geocities.com/top-bp"
+	}
+  ],
+  "lifecycle": {
+    "version": "<version of lifecycle in builder>",
+    "apis":  {
+        "buildpack": {
+          "deprecated": ["0.1"],
+          "supported": ["1.2", "1.3"]},
+        "platform": {
+          "deprecated": [],
+          "supported": ["2.3", "2.4"]}
+    }
+  },
+  "createdBy": {
+    "name": "<creator of builder>", 
+    "version": "<builder version>"
+  }
+}
+```
+
+#### Layers
+The `io.buildpacks.buildpack.layers` data should look like:
+```json
+{
+  "<buildpack id>": {
+    "<buildpack version>": {
+      "api": "<buildpack API>",
+      "order": [
+        {
+          "group": [
+            {
+              "id": "<inner buildpack>",
+              "version": "<inner bulidpacks version>"
+            }
+          ]
+        }
+      ],
+      "layerDiffID": "sha256:test.nested.sha256",
+	  "homepage": "http://geocities.com/top-bp"
+    }
+  },
+  "<inner buildpack>": {
+    "<inner buildpacks version>": {
+      "api": "<buildpack API>",
+      "stacks": [
+        {
+          "id": "<stack ID buildpack supports>",
+          "mixins": ["<list of mixins required>"]
+        }
+      ],
+      "layerDiffID": "sha256:test.bp.one.sha256",
+	  "homepage": "http://geocities.com/cool-bp"
+    }
+  }
+}
+```
+
+## Spec Additions
+### [Buildpack specification][buildpacks-spec]
+Builders don't seem immediately relevant to the buildpack specification; as such, I wouldn't add mention of builders here. 
+
+### [Platform specification][platform-spec]
+We can add a note, clarifying that in platforms that use builders, the Stack's build image is the core of the builder. 
+
+### [Distribution specification][distribution-spec]
+We should have a rigorous definition of the builder here, with a description of all the files and metadata necessary and allowed. 
+
+# Drawbacks
+[drawbacks]: #drawbacks
+N/A
+
+# Alternatives
+[alternatives]: #alternatives
+Specifying the builder is something that, to my mind, should definitely done; the question is, how. We could have an entirely separate `extension spec`, defining builders and how they should be used. 
+ > Given that one of the key uses of builders is in distributing buildpacks, I thought it should at least be discussed in some depth in the [distribution spec][distribution-spec], and given that the buildpackage makeup is described there, I thought it appropriate to just describe it fully in the distribution spec. 
+
+# Prior Art
+[prior-art]: #prior-art
+Some similar PRs that led to the development of specifications are:
+* [Distribution Specification][distrib-spec-rfc]
+* [Service Binding Specification][service-binding-rfc]
+
+# Unresolved Questions
```

Comment:
Should we also define the schema of the `builder.toml`?

Reply:
It could make sense to do so - but on the other hand, `builder.toml` is used in pack, but should it be standardized for other platforms as well? It feels like more of an implementation detail of how pack creates builders.

Reply:
Seemed like consensus in Working Group was not to define it here.